components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2157,6 +2158,7 @@ public ServiceProvider getApplication(int applicationId) throws IdentityApplicat
                     applicationId, connection, tenantID);
             serviceProvider.setRequestPathAuthenticatorConfigs(requestPathAuthenticators);
 
+            removeSpPropertiesFromAdditionalProps(propertyList);
```

Comment:
Shall we remove these in the places where we were getting these configs?  In the same way, we removed for getTemplateId.

Reply:
Tried that for three of the fields (skip login/ logout consent and roles). Throwing concurrent handling exception though removed with iterator.